yin/rubysonar-master/src/test/java/org/yinwang/rubysonar/_Test.java

```diff
@@ -0,0 +1,84 @@
+package org.yinwang.rubysonar;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.ArrayList;
+
+public class _Test {
+
+@Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testSame() {
+    Assert.assertTrue(_.same(null, null));
+    Assert.assertFalse(_.same(1, 0));
+  }
+
+  @Test
+  public void testGetQnameParent() {
+    Assert.assertEquals("", _.getQnameParent(null));
+    Assert.assertEquals("", _.getQnameParent("1234"));
+    Assert.assertEquals(
+      "...............",
+      _.getQnameParent("................?????????????"));
+  }
+
+  @Test
+  public void testMainName() {
+    thrown.expect(NullPointerException.class);
+    _.mainName(null);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @Test
+  public void testArrayToString() {
+    // Arrange
+    final ArrayList<String> strings = new ArrayList<String>();
+    Assert.assertEquals("", _.arrayToString(strings));
+
+    strings.add("foo");
+    Assert.assertEquals("foo\n", _.arrayToString(strings));
```

Comment:
I would add one more to show that after each object a new line is added rather than the end of the arraylist.. "foo\nbar\n"